### PR TITLE
Copy build_tar code from rules_pkg locally

### PR DESCRIPTION
**Pull request by cristifalcas**
Newer rules_pkg version are not compatible anymore with distroless. This means that people can't update to a newer rules_pkg.

This PR adds the last code working for `build_tar` from rules_pkg to this repo.

### CODE REVIEW

File: tools/BUILD [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+py_library(
+    name = "archive",
+    srcs = [
+        "__init__.py",
+        "archive.py",
+    ],
+    srcs_version = "PY3",
+    visibility = [
+        "//experimental:__pkg__",
+        "//tests:__pkg__",
```
Comment (loosebazooka) [inline tools/BUILD:13]: I don't know if this visibility makes sense here? we don't have a `tests` at root?

**Review by loosebazooka — COMMENTED** [review]
Thanks for doing this.
1. Can you rename the `tools` directory to something more descriptive. Especially if we don't intend on adding too many more tools here.
2. Can you include a readme in the directory of what files you copied from where?
3. Were no tests available to copy over?

File: tools/BUILD (hunk 1 above)
Comment (cristifalcas) [reply tools/BUILD]: blind copy pasta. Fixed now

**Comment by cristifalcas** [comment]
> Thanks for doing this.
> 
> 1. Can you rename the `tools` directory to something more descriptive. Especially if we don't intend on adding too many more tools here.
> 2. Can you include a readme in the directory of what files you copied from where?
> 3. Were no tests available to copy over?

1. done
2. done
3. no tests exists for this specific part

**Comment by loosebazooka** [comment]
autoformat the pycode?

**Comment by cristifalcas** [comment]
> I think you need to format the python code somehow?

Should be ok now

**Comment by loosebazooka** [comment]
pylint is still complaining a bit

**Comment by cristifalcas** [comment]
test are passing now on my fork

**Comment by cristifalcas** [comment]
drat! It failed from some TODO comments, from what I can see. Removed, please run the tests again

**Pull request merged**